src/context.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+
+
+const initialArtboardState = {
+  width: 0,
+  height: 0,
+  scale: 1,
+  fontScale: 1,
+};
+
+const ArtboardContext = React.createContext({
+  state: initialArtboardState,
+});
+
+export const useWindowDimensions = () => {
+  const { state } = React.useContext(ArtboardContext);
+  const { width, height } = state || {};
+
+  return { width, height };
+}
+
+const ArtboardPropTypes = {
+  viewport: PropTypes.shape({
+    width: PropTypes.number,
+    height: PropTypes.number,
+    name: PropTypes.string,
+  }),
+  children: PropTypes.node,
+};
+
+type ArtboardProps = PropTypes.InferProps<typeof ArtboardPropTypes>;
+
+export const ArtboardProvider = ({ children, viewport }: ArtboardProps & { children: any }) => {
```

Comment:
So I'm not sure it's enough to only consider the viewport. Most of the time people won't specify it, and only use the style's width (like in the basic example https://github.com/airbnb/react-sketchapp/blob/master/examples/basic-setup/src/my-command.js#L45-L51).

So IMO it'd be useful to fallback to the style dimension if we have no viewport (even if we have only the width because it's probably the main use case).

Reply:
Good point, am falling back to `style.width` and `style.height`. Is it worth doing a `console.warn` in the hook, if the `viewport` is missing, prompting them to use the `<Artboard viewport={}` prop? Incase they get confused about why the height is the full page height rather than viewport height?

Reply:
Nah I think that what you actually expect: just the size of the artboard. I actually didn't even think about the viewport prop before your PR.